### ISSUE TITLE
[`flake8-slots`] Rule `SLOT002` doesn't take `typing.NamedTuple` into account

### ISSUE THREAD

**Issue by autinerd** [astral-sh/ruff#10805]
Hello :blush: 

While adding the `SLOT` rules to Home Assistant, I found that the `SLOT002` doesn't take subclasses of `typing.NamedTuple` in account, although they are the typed version of `collections.namedtuple`.

Thanks in advance!

**Comment by AlexWaygood** [comment]
Hi, thanks for opening the issue! Can you give a minimal code snippet as an example of where you think this rule should be emitting an error but isn't currently?

**Comment by AlexWaygood** [comment]
Ah, do you mean something like this?

```py
from typing import NamedTuple

class Foo(NamedTuple("Foo", [("x", int)])):
    pass
```

It's unusual to inherit from a call-based `typing.NamedTuple` rather than to do something like

```py
from typing import NamedTuple

class Foo(NamedTuple):
    x: int
```

But I suppose if you _do_ inherit from a call-based `typing.NamedTuple`, we probably _should_ emit the diagnostic the same as if it were a `collections.namedtuple` (and we currently don't)./

**Comment by autinerd** [comment]
Oh, I didn't know that this example:

```
from typing import NamedTuple

class Foo(NamedTuple):
    x: int
```

will have the `__slots__` automatically created, then it is not needed of course, thanks.

And I didn't know about the call-based NamedTuple as well :sweat_smile: (it is not used in the Home Assistant codebase), but that would be a good idea to add.
